include/GafferSceneUI/ContextAlgo.h

```diff
@@ -74,6 +74,10 @@ void expand( Gaffer::Context *context, const GafferScene::PathMatcher &paths, bo
 /// Returns a new PathMatcher containing the new leafs of this expansion.
 GafferScene::PathMatcher expandDescendants( Gaffer::Context *context, const GafferScene::PathMatcher &paths, const GafferScene::ScenePlug *scene, int depth = Imath::limits<int>::max() );
 
+/// Removes paths from the current expansion.
+/// Returns a new PathMatcher containing the removed paths.
+GafferScene::PathMatcher collapse( Gaffer::Context *context, const GafferScene::PathMatcher &paths );
```

Comment:
Reverted in 2109160bbc1c85b27430dd0e892ac440f03839fc and 7d2ee7ea2f6346c39a00b36585d0d8878434f34c. I've kept e179961acd6d12d773a641818b5ff6919c651089 because it otherwise `SceneView::collapseSelection()` was untested.